includes/modules/pages/password_forgotten/header_php.php

```diff
@@ -66,14 +66,26 @@
 
     // handle 3rd-party integrations
     $zco_notifier->notify('NOTIFY_PASSWORD_FORGOTTEN_CHANGED', $email_address, $check_customer->fields['customers_id'], $new_password);
+      if (PADSS_ADMIN_SESSION_TIMEOUT_ENFORCED) {
+          $messageStack->add_session('login', SUCCESS_PASSWORD_SENT, 'success');
+      } else {
+          $messageStack->add_session('login', SUCCESS_PASSWORD_SENT_CONFIRM, 'success');
+      }
+      zen_redirect(zen_href_link(FILENAME_LOGIN, '', 'SSL'));
 
   } else {
     $zco_notifier->notify('NOTIFY_PASSWORD_FORGOTTEN_NOT_FOUND', $email_address);
+      if (PADSS_ADMIN_SESSION_TIMEOUT_ENFORCED) {
+          $messageStack->add_session('login', SUCCESS_PASSWORD_SENT, 'success');
+          zen_redirect(zen_href_link(FILENAME_LOGIN, '', 'SSL'));
+      } else {
+          $messageStack->add_session('password_forgotten', EMAIL_ADDRESS_NOT_FOUND);

```

Comment:
Does this `add_session` actually work as expected? Given that you're bypassing the `zen_redirect` here, you probably need to use `add` instead of `add_session` (since `add_session` is for making the message show up on the "next" page, not the current page).

I suggest testing the use of `add` vs `add_session`, to determine whether you're getting the desired result.